src/Common/AsynchronousMetrics.cpp

```diff
@@ -882,9 +884,10 @@ void AsynchronousMetrics::update(TimePoint update_time)
         }
     }
 
-    if (cgroupmem_limit_in_bytes && cgroupmem_usage_in_bytes)
+    if (cgroupmem_limit_in_bytes || cgroupmem_usage_in_bytes || cgroupmem_v2_limit_in_bytes || cgroupmem_v2_usage_in_bytes)
```

Comment:
Pointer dereferences are inconsistent with this check.

Reply:
now it's completely separated with shared private function